### ISSUE TITLE
Add DNS-backed Registry

### ISSUE THREAD

**Issue by richardartoul** [richardartoul/nola#38]
NOLA's current implementation requires an external registry to function. The external registry can be backed by FoundationDB, local memory, or (soon) Postgres. A strongly consistent metadata store is required to back the Registry to Ensure NOLA's linearizability / correctness guarantees.

However, there are use-cases where NOLA's abstraction would still be very useful, but where deploying an external metadata store is overkill. For example, NOLA could be used to implement something like [groupcache](https://github.com/golang/groupcache). However, deploying Postgres or FoundationDB just to power an in-memory cache with built in loading deduplication feels like overkill.

For scenarios like that, it would be nice to be able to leverage NOLA with no dependencies except DNS and a functioning network. I think this should be possible by making a DNS backed implementation of the Registry abstraction that doesn't implement many of its methods (like per-actor KV storage, etc) and instead focuses on discovering peers using DNS and then making it so that `EnsureActivations` is implemented by building a consistent hashing ring over the DNS-discovered IP addresses.

Basically there are scenarios where NOLA's routing + discovery abstractions are useful (any scenario where consistent-hashing routing might be useful) but the correctness requirements are not that strict, so we should have a way to power those use-cases without having to deploy an external store. This would also lower the barrier to adopting NOLA by letting people experiment with it in their own environments using DNS (even though the implementation would not be linerizable) and then "upgrade" to a linearizable implemention with built in durable KV storage once they're ready.
